pkg/kubelet/cm/topologymanager/policy_single_numa_node.go

```diff
@@ -158,7 +127,8 @@ func (p *singleNumaNodePolicy) mergeProvidersHints(providersHints []map[string][
 }
 
 func (p *singleNumaNodePolicy) Merge(providersHints []map[string][]TopologyHint) (TopologyHint, lifecycle.PodAdmitResult) {
-	hint := p.mergeProvidersHints(providersHints)
+	filteredHints := filterProvidersHints(providersHints)
+	hint := p.mergeProvidersHints(filteredHints)
 	admit := p.canAdmitPodResult(&hint)
```

Comment:
Can you move the call to `filterSingleNumaHints()` out to here so that all the filtering happens in one place? Something like:
```
filteredHints := filterProvidersHints(providersHints)
filteredHints = filterSingleNumaHints(filteredHints)
hint := p.mergeProvidersHints(filteredHints)
admit := p.canAdmitPodResult(&hint)
```

Reply:
After doing this, in theory, this whole function could be reused across all of the policies, if we standardized on `TopologyHint{nil, bestHint.Preferred}` referring to a "don't care" hint. If we did this (in the body of `mergePermutation()`, for example), then we could remove the following from here and have a common `mergeProvidersHints()` function:
```
        if bestHint.NUMANodeAffinity.IsEqual(defaultAffinity) {
                bestHint = TopologyHint{nil, bestHint.Preferred}
        }
```

Reply:
Looks like both of these were actually address in later commits that I hadn't gotten to yet.

/lgtm
/approve